### PR TITLE
Feature/sim 2052 fewer filters

### PR DESCRIPTION
**Pull request by yoachim**
Added kwarg so one can only run the pre-computed magnitude interpolation on a subset of filters rather than the default of all 6.  Haven't actually timed it to see if there is significant speedup.

### CODE REVIEW

**Comment by rhiannonlynne** [comment]
So -- not a huge speedup (for 1000 single-pointing skybrightnesses, it goes from 1.76s to 1.39s to do the actual sky brightness calculation part, using setParams). 
I did see that I could pick up a bit more speed by not converting the result of returnMags into a structured array (useful, if there is only one filter being used) (1.18s for 1000 pointings). This however is not great as formatting of the result depends on how the method is set up. 
Returning it as a dictionary instead of a structured array still results in a speedup (1.21s for 1000 pointings, with a single filter, 1.41s for all filters). Could the dictionary return values work? 
(even if you can't use the output as a dictionary, but need a structured array, I think using np.swapaxes(0, 1) is still faster than np.transpose() on the array and is equivalent here). 

I'm a little confused as to what's actually being read from the files when 'mags' is True. Are you actually reading spectra, and then just converting back to a mag value before returning to the user? It looks like this is what's going on, but I suspect what's actually happening is that you have a single flux value which represents the flux through the entire LSST bandpass, instead of a full wavelength spectrum (when mags = True). But I can really only guess that because I think I know what you did in the skybrightness data repo. It's a bit hard to follow, as it happens without much description and with hard-to-read python in the 'loadSpec' function and then just gets used in an identical way in 'weighting', so some more comments or documentation would be useful.

I'm a little curious why (in skyModel) you don't make interpSky a private method, and have that called directly from returnMags and returnWaveSpec? (am I missing some reason you'd want users to be able to call interpSky?)

**Comment by rhiannonlynne** [comment]
Just a note - the documentation for returnMags implies you can set your own bandpasses (multiples), but actually the code doesn't iterate over bandpass, so it should be just a single bandpass at a time. And if it's used, then the return value of 'returnMags' is different than if you had self.mags=True (a single array, vs. a structured array). 

Note that 'bandpass' has a "bandpassname" attribute, which matches the filename if read from a file -- although it could be just "fromArrays" if set directly from an array (and thus could potentially be duplicated). You could try using that so that you could still set up a structured array.  (or swap to dictionaries .. although in either case, I guess the user might not realize that 'bandpassname' is the key they should be looking for (although this could be mentioned in the docstring).

**Comment by rhiannonlynne** [comment]
I also pulled in changes from a branch I was working on to fix a few problems in setParams (being able to override alt values) and to allow single-pointings. The feature/single-pointings branch becomes obsolete here.

**Comment by yoachim** [comment]
ugh, this whole repo needs lots more documentation.

Dictionaries should work. I made it a kwarg option.

Yup, interpSky should be a private method.  I call it earlier than returnMags so that one can loop over multiple filters and not re-compute the spectra each time.  

The LSST magnitudes are already computed and saved in the npz files, so when mags=True, it just reads everything in like normal but skips reading in the full spectra since they won't be needed.

File: python/lsst/sims/skybrightness/skyModel.py [hunk 1]
```diff
@@ -591,46 +591,42 @@ def returnWaveSpec(self):
...
+                maxWave = bandpasses[key].wavelen[isThrough].max()
+                inBand = np.where((self.wave >= minWave) & (self.wave <= maxWave))
+                for i, ra in enumerate(self.ra):
+                    # Check that there is flux in the band, otherwise calcMag fails
+                    if np.max(self.spec[i, inBand]) > 0:
+                        tempSed.setSED(self.wave, flambda=self.spec[i, :])
+                        mags[i] = tempSed.calcMag(bandpasses[key])
+                # Mask out high airmass
+                mags[self.mask] *= np.nan
+                magsBack[key] = mags.copy()
```
Comment (rhiannonlynne) [inline python/lsst/sims/skybrightness/skyModel.py]: I don't think you actually need to copy this here, do you? the mags array itself would disappear after returning from the method, unless you add the reference. So you can just reference it directly, unless you think something would change mags?

File: tests/testSkyModel.py [hunk 2]
```diff
@@ -160,7 +158,7 @@ def test90Deg(self):
         sm = self.sm_mags
         sm.setRaDecMjd(0., 90., mjd, degrees=True, azAlt=True)
         mags = sm.returnMags()
-        for key in mags.dtype.names:
+        for key in mags.keys():
```
Comment (rhiannonlynne) [inline tests/testSkyModel.py]: I think to be easier to translate to python3, you can just do 'for key in mags:'  (the iterator over mags is over the keys).

File: python/lsst/sims/skybrightness/skyModel.py (hunk 1 above)
Comment (yoachim) [inline python/lsst/sims/skybrightness/skyModel.py]: yup, don't actually need to copy.

File: tests/testSkyModel.py (hunk 2 above)
Comment (yoachim) [inline tests/testSkyModel.py]: I tend to put the .keys() in there for my own readability since I can never remember what gets iterated. Good point about Python 3.

**Comment by rhiannonlynne** [comment]
Looks fine to me. yeah ... more documentation would be great. Maybe another ticket.
